src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -268,6 +268,11 @@ class SparqlQleverVisitor {
 
   Triples visit(Parser::TriplesTemplateContext* ctx);
 
+  // Limit the variables in the filter to the ones that are already known
```

Comment:
|Be more precise: you limit the selected variables inside the EXISTS expressions that are contained in the `filter` etc.